<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>720</width>
    <height>550</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>720</width>
    <height>550</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Декартов лист</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="GLWidget" name="glWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>700</width>
      <height>400</height>
     </rect>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>420</y>
      <width>351</width>
      <height>121</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="InfoTab">
     <attribute name="title">
      <string>Информация</string>
     </attribute>
     <widget class="QLabel" name="EqLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>161</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap>equation.png</pixmap>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>10</y>
        <width>141</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Информация</string>
      </property>
      <widget class="QWidget" name="formLayoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>121</width>
         <height>51</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_1">
        <item row="0" column="0">
         <widget class="QLabel" name="SquareLabel">
          <property name="text">
           <string>Площадь:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="Square">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="LengthLabel">
          <property name="text">
           <string>Длина:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="Length">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>180</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Формула</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="ParamTab">
     <attribute name="title">
      <string>Параметры</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_5">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>10</y>
        <width>101</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Параметры</string>
      </property>
      <widget class="QWidget" name="formLayoutWidget_3">
       <property name="geometry">
        <rect>
         <x>13</x>
         <y>30</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_4">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="AParamLabel">
          <property name="text">
           <string>a:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="AParamEdit">
          <property name="minimum">
           <double>-9.000000000000000</double>
          </property>
          <property name="maximum">
           <double>9.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>221</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Доп. построения</string>
      </property>
      <widget class="QWidget" name="formLayoutWidget_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>202</width>
         <height>60</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <property name="horizontalSpacing">
         <number>0</number>
        </property>
        <property name="verticalSpacing">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QCheckBox" name="DrawTangentLine">
          <property name="text">
           <string>касательная и нормаль</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Точка касания:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QCheckBox" name="DrawFlower">
          <property name="text">
           <string>цветок</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <number>-80</number>
          </property>
          <property name="maximum">
           <number>220</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="TransoformationTab">
     <attribute name="title">
      <string>Преобразования</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>131</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Коорд. узловой точки</string>
      </property>
      <widget class="QWidget" name="formLayoutWidget_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>111</width>
         <height>51</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_5">
        <item row="0" column="0">
         <widget class="QLabel" name="XLabel_2">
          <property name="text">
           <string>X</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="YLabel_2">
          <property name="text">
           <string>Y</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="XEdit_2">
          <property name="minimum">
           <number>-18</number>
          </property>
          <property name="maximum">
           <number>18</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="YEdit_2">
          <property name="minimum">
           <number>-9</number>
          </property>
          <property name="maximum">
           <number>9</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_7">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>10</y>
        <width>120</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Поворот</string>
      </property>
      <widget class="QWidget" name="formLayoutWidget_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>101</width>
         <height>22</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_6">
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Угол</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="AngleEdit">
          <property name="minimum">
           <number>-360</number>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
   <slots>
    <slot>TranslateX(int)</slot>
    <slot>TranslateY(int)</slot>
    <slot>Rotate(int)</slot>
    <slot>showFlower(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>XEdit_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>TranslateX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>482</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>YEdit_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>TranslateY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>508</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AngleEdit</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glWidget</receiver>
   <slot>Rotate(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>492</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DrawFlower</sender>
   <signal>toggled(bool)</signal>
   <receiver>glWidget</receiver>
   <slot>showFlower(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>519</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DrawFlower</sender>
   <signal>toggled(bool)</signal>
   <receiver>DrawTangentLine</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>519</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>481</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DrawTangentLine</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>254</x>
     <y>492</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DrawFlower</sender>
   <signal>toggled(bool)</signal>
   <receiver>DrawTangentLine</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>519</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>481</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
